From 633fac9efebfeb56480990effa0542eb39e0f7fb Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Fri, 2 Nov 2012 18:13:23 -0600
Subject: [PATCH 020/248] FogBugz #83095: Add code to determine which hardware
 linux is running on

Use device tree entries to determine which hardware linux is running on.
For this patch, its only make the appropriate change to CPU1 start addr,
but it also lays the foundation on how this can be done.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
---
 arch/arm/mach-socfpga/core.h    |  2 --
 arch/arm/mach-socfpga/socfpga.c | 19 +++++++++++++++++--
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-socfpga/core.h b/arch/arm/mach-socfpga/core.h
index fedf1b4..61b0181 100644
--- a/arch/arm/mach-socfpga/core.h
+++ b/arch/arm/mach-socfpga/core.h
@@ -40,8 +40,6 @@ extern void socfpga_init_clocks(void);
 
 extern unsigned long cpu1start_addr;
 
-#define SOCFPGA_SCU_VIRT_BASE   0xfffec000
-
 #define SOCFPGA_SCU_VIRT_BASE	0xfffec000
 #define SOCFPGA_SDMMC_BASE	0xff704000
 
diff --git a/arch/arm/mach-socfpga/socfpga.c b/arch/arm/mach-socfpga/socfpga.c
index 46542f7..e4be3aa 100644
--- a/arch/arm/mach-socfpga/socfpga.c
+++ b/arch/arm/mach-socfpga/socfpga.c
@@ -32,8 +32,8 @@
 void __iomem *socfpga_scu_base_addr = ((void __iomem *)(SOCFPGA_SCU_VIRT_BASE));
 void __iomem *sys_manager_base_addr;
 void __iomem *rst_manager_base_addr;
-void __iomem *clk_mgr_base_addr;
-unsigned long cpu1start_addr;
+
+unsigned long	cpu1start_addr;
 
 static const struct of_dev_auxdata socfpga_auxdata_lookup[] __initconst = {
 #ifdef CONFIG_MMC_DW
@@ -73,6 +73,16 @@ static void __init socfpga_scu_map_io(void)
 	iotable_init(&scu_io_desc, 1);
 }
 
+static void __init init_socfpga_vt(void)
+{
+	cpu1start_addr = 0x10;
+}
+
+static void __init init_socfpga(void)
+{
+	cpu1start_addr = 0xc4;
+}
+
 static void __init enable_periphs(void)
 {
 	/* Release all peripherals from reset.*/
@@ -120,6 +130,11 @@ static void __init socfpga_init_irq(void)
 {
 	irqchip_init();
 	socfpga_sysmgr_init();
+
+	if (of_machine_is_compatible("altr,socfpga-vt"))
+		init_socfpga_vt();
+	else
+		init_socfpga();
 }
 
 static void socfpga_cyclone5_restart(enum reboot_mode mode, const char *cmd)
-- 
1.9.1

